src/clr-angular/layout/tabs/tabs.ts

```diff
@@ -51,13 +56,35 @@ import { ClrCommonStrings } from '../../utils/i18n/common-strings.interface';
         </ng-container>
     `,
   providers: [IfActiveService, IfOpenService, TabsService, TABS_ID_PROVIDER],
+  host: {
+    '[class.tabs-vertical]': 'isVertical()',
+  },
 })
-export class ClrTabs implements AfterContentInit {
-  @ContentChildren(ClrTabLink, { descendants: true })
-  tabLinkDirectives: QueryList<ClrTabLink>;
+export class ClrTabs implements AfterContentInit, OnDestroy {
+  private subscription: Subscription;
+
+  @Input('clrTabsLayout')
+  set layout(layout: TabsLayout) {
+    if (!Object.values(TabsLayout).includes(layout)) {
+      throw `Unsupported layout ${layout}`;
```

Comment:
I don't think we want to throw an error here, I think we just want to default to horizontal. @youdz?

Reply:
You're right, I didn't pick up on this but some people don't have the cleanest lifecycles and we do receive the occasional `undefined` before the actual correct value. This would throw errors too often. Just rely on Typescript, you already made the argument of type `TabsLayout`, so Typescript will throw an error in AoT compilation if they give us something else.

TL;DR: Keep the if, but just ignore instead of throwing an error.

Reply:
Done